core/libs/oauth2.0/storage/pdo.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * Projects4Me Copyright (c) 2017. Licensing : http://legal.projects4.me/LICENSE.txt. Do not remove this line
+ */
+
+namespace Gaia\Libraries\Oauth\Storage;
+
+/**
+ * This class extends Pdo class provided by bshaffer/oauth2-server-php library. In this class we're overriding
+ * checkPassword and hashPassword method in order to follow our hashing functionality. By default hashPassword method
+ * use "sha1" algorithm in order to match user password. But we have changed the encryption of user password using
+ * "bcrypt" alogrithm to add more security. So now we need to match the hash generated by bcrypt algorithm. That's why
+ * we needed to override these methods.
+ * 
+ * @author Rana Nouman <ranamnouman@gmail.com>
+ */
+class Pdo extends \OAuth2\Storage\Pdo
+{
+
+    /**
+     * This function checks user password.
+     *
+     * @param array $user
+     * @param string $password
+     * @return string
+     */
+    public function checkPassword($user, $password)
```

Comment:
Is there a need to override this function?

Reply:
Yes, because **password_verify()** function in the **hashPassword()** method requires user's password hash stored in the database in order to compare it with the password input by user to get access token. So that information (password) of user is available inside **checkPassword()** method. That's why we need to override this method.